Source/Core/DolphinQt/GameList/GameListModel.cpp

```diff
@@ -140,6 +141,26 @@ QVariant GameListModel::data(const QModelIndex& index, int role) const
     if (role == Qt::DisplayRole || role == Qt::InitialSortOrderRole)
       return QString::fromStdString(game.GetFileName());
     break;
+  case COL_FILE_PATH:
+    if (role == Qt::DisplayRole || role == Qt::InitialSortOrderRole)
+    {
+      QString file_path = QDir::cleanPath(QString::fromStdString(game.GetFilePath()));
+      for (QString dir : Settings::Instance().GetPaths())
+      {
+        dir = QDir::cleanPath(dir);
+        if (file_path.startsWith(dir))
+        {
+          int path_index = dir.lastIndexOf(QLatin1Char('/'), -2);
+          if (path_index > -1)
+          {
+            file_path = file_path.mid(path_index + 1);
+            break;
+          }
+        }
+      }
+      return QDir::toNativeSeparators(file_path);
+    }
+    break;
```

Comment:
It might not matter nowadays, but keep in mind that this function is called *a lot*, many times for each item. At some point we'll need some caching. I don't know whether we need caching yet. Maybe make a few hundred copies of a .dol file and put them in a deep path. Can you still scroll without maxing your CPU?

Reply:
It's a simple iteration over a string replacing backslashes with slashes, it would hardly put a dent in the CPU's performance

Reply:
Per game, it's called once per game directory preset until a match is found, and then only once more.

Reply:
Maybe the function to store/get preset paths should be platform independent, so it's only called once per directory addition, if performance is a concern

Reply:
I mean the function `data()`.

Reply:
Ok

Reply:
That's outside of the scope of this PR, considering this is my first, but perhaps I could give caching a shot for another PR?

Reply:
imo it should be handled separately, yes. I am pretty sure this could be cached by Qt itself by setting column's display role appropriately at creation time.